### PR TITLE
[cov-test] speed up cov test

### PR DESCRIPTION
**Pull request by jiangying000**
## Pull request type

speed up cov test

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #3699 

## What is the new behavior?

-
-
-

### CODE REVIEW

File: scripts/coverage_report.sh [hunk 1]
```diff
@@ -65,9 +65,22 @@ if [ $SKIP_PROMPTS -eq 0 ]; then
...
+# export RUSTC_BOOTSTRAP=1
+# export CARGO_INCREMENTAL=0
+
+# env generated by `cargo llvm-cov show-env` on current CI server
+# ----------------------------------------
+# RUSTFLAGS=" -C instrument-coverage --cfg coverage --cfg trybuild_no_target"
+# LLVM_PROFILE_FILE="/runner/_work/starcoin/starcoin/target/starcoin-%m.profraw"
+# CARGO_INCREMENTAL="0"
+# RUST_TEST_THREADS="1"
+# CARGO_LLVM_COV_TARGET_DIR="/runner/_work/starcoin/starcoin/target"
```
Comment (jiangying000) [inline scripts/coverage_report.sh:78]: not tested which of these env makes test faster, but it does works

Comment (jiangying000) [inline scripts/coverage_report.sh:78]: not tested which of these env makes the coverage test faster, but it does work

Comment (jiangying000) [reply scripts/coverage_report.sh:78]: @taiki-e could you take a look ?

Comment (taiki-e) [reply scripts/coverage_report.sh:78]: `-Zprofile`, `-Ccodegen-units=1`, and `-Clink-dead-code` are unneeded for cargo-llvm-cov and removal of them may improve performance.

File: scripts/coverage_report.sh [hunk 2]
```diff
@@ -65,9 +65,22 @@ if [ $SKIP_PROMPTS -eq 0 ]; then
...
+# ----------------------------------------
+
+# Set the environment variables needed to get coverage.
+source <(cargo llvm-cov show-env --export-prefix)
```
Comment (taiki-e) [inline scripts/coverage_report.sh:82]: This only sets the environment variables that cargo-llvm-cov normally sets for cargo, and seems unnecessary for your use case.

File: scripts/coverage_report.sh (hunk 1 above)
Comment (jiangying000) [reply scripts/coverage_report.sh:78]: thanks for this info

File: scripts/coverage_report.sh (hunk 2 above)
Comment (jiangying000) [reply scripts/coverage_report.sh]: thanks for this info

Comment (jiangying000) [reply scripts/coverage_report.sh]: I cleaned up the script according to your suggestions, and it works fine

Can I improve more in our use case, or could you help approve this change ?

thanks

**Review by jolestar — APPROVED** [review]

**Pull request merged**